src/cloud_provider/aws/kubernetes/mod.rs

```diff
@@ -119,1675 +109,1168 @@ pub struct Options {
+            event_details,
+            EventMessage::new(
+                "Error trying to get kubernetes events".to_string(),
+                Some(err.message(ErrorMessageVerbosity::FullDetails)),
+            ),
+        )),
+    };
+
+    Ok(())
+}
 
-        Ok(())
-    }
+fn upgrade_error(kubernetes: &dyn Kubernetes) -> Result<(), EngineError> {
+    kubernetes.logger().log(EngineEvent::Warning(
+        kubernetes.get_event_details(Stage::Infrastructure(InfrastructureStep::Upgrade)),
+        EventMessage::new_from_safe(format!("{}.upgrade_error() called.", kubernetes.kind())),
+    ));
 
-    fn downgrade(&self) -> Result<(), EngineError> {
-        Ok(())
-    }
+    Ok(())
+}
 
-    fn downgrade_error(&self) -> Result<(), EngineError> {
-        self.logger().log(EngineEvent::Warning(
-            self.get_event_details(Stage::Infrastructure(InfrastructureStep::Downgrade)),
-            EventMessage::new_from_safe("EKS.downgrade_error() called.".to_string()),
-        ));
+fn downgrade() -> Result<(), EngineError> {
+    Ok(())
+}
 
-        Ok(())
-    }
+fn downgrade_error(kubernetes: &dyn Kubernetes) -> Result<(), EngineError> {
+    kubernetes.logger().log(EngineEvent::Warning(
+        kubernetes.get_event_details(Stage::Infrastructure(InfrastructureStep::Downgrade)),
+        EventMessage::new_from_safe(format!("{}.downgrade_error() called.", kubernetes.kind())),
+    ));
 
-    fn pause(&self) -> Result<(), EngineError> {
-        let event_details = self.get_event_details(Stage::Infrastructure(InfrastructureStep::Pause));
-        let listeners_helper = ListenersHelper::new(&self.listeners);
+    Ok(())
+}
 
-        self.send_to_customer(
-            format!("Preparing EKS {} cluster pause with id {}", self.name(), self.id()).as_str(),
-            &listeners_helper,
-        );
+fn pause(
+    kubernetes: &dyn Kubernetes,
+    template_directory: &str,
+    aws_zones: &[AwsZones],
+    node_groups: &[NodeGroups],
+    options: &Options,
+) -> Result<(), EngineError> {
+    let event_details = kubernetes.get_event_details(Stage::Infrastructure(InfrastructureStep::Pause));
+    let listeners_helper = ListenersHelper::new(kubernetes.listeners());
+
+    kubernetes.send_to_customer(
+        format!(
+            "Preparing {} {} cluster pause with id {}",
+            kubernetes.kind(),
+            kubernetes.name(),
+            kubernetes.id()
+        )
+        .as_str(),
+        &listeners_helper,
+    );
 
-        self.logger().log(EngineEvent::Info(
-            self.get_event_details(Stage::Infrastructure(InfrastructureStep::Pause)),
-            EventMessage::new_from_safe("Preparing EKS cluster pause.".to_string()),
-        ));
+    kubernetes.logger().log(EngineEvent::Info(
+        kubernetes.get_event_details(Stage::Infrastructure(InfrastructureStep::Pause)),
+        EventMessage::new_from_safe(format!("Preparing {} cluster pause.", kubernetes.kind())),
```

Comment:
OPT: `kubernetes.kind()` shouldn't be needed here. Logging system does add this info, no need to add it.

 `EventMessage::new_from_safe("Preparing cluster pause.".to_string())`
 
 Should do the trick.

Reply:
ack